internal/model/rwfolder.go

```diff
@@ -492,6 +494,18 @@ func (p *rwFolder) pullerIteration(ignores *ignore.Matcher) int {
 		return true
 	})
 
+	// Check if we are able to store all files on disk
+	if pullFileSize > 0 {
+		folder, ok := p.model.cfg.Folders()[p.folder]
+		if ok {
+			diskSize, diskFree, diskErr := osutil.DiskUsage(p.folder)
+			if diskErr == nil && (diskSize-diskFree)-pullFileSize < (diskSize/100)*int64(folder.MinDiskFreePct) {
```

Comment:
This is too much arithmetic in the if clause, I find it hard to follow and I'm not sure it's correct. The current amount of used space (size - free), minus the space we need, less than the percentage specified?

I would expect the checkFolderHealth thing to take care of the percentage, and here just check if we have space for the files we are about to pull, i.e. something like diskFree < pullFileSize?